### ISSUE TITLE
getblocktemplate not available for up to 20 minutes after starting

### ISSUE THREAD

**Issue by flound1129** [dashpay/dash#1738]
GBT should be fully functional once the blockchain is synced.  But for up to 20 minutes after a node starts, GBT fails with 
```
error code: -10
error message:
Dash Core is syncing with network...
```

**Comment by schinzelh** [comment]
Hi,

the message is not only referring to blochain sync, but also to governance object sync, which may take up to 60 mins after restart. You can check the status of your wallet with

ˋmnsync statusˋ

as soon the the status is complete/999 GBT will be available.

**Comment by flound1129** [comment]
Why does the governance object need to be synced in order to return a block template?

Just return an empty template or exclude transactions related to governance until the sync is complete.

**Comment by flound1129** [comment]
IIRC masternodes also can't have more than an hour of downtime before losing their spot in the payment queue, so this is also an important issue for masternode operators.

**Comment by UdjinM6** [comment]
>GBT should be fully functional once the blockchain is synced.

Unlike in Bitcoin, in Dash you can't construct correct block having only blockchain data, you need at least masternode payee and some times all governance info. But yes, current situation is a bit suboptimal and can be improved, please see #1739

**Comment by flound1129** [comment]
This looks to be exactly what I need, thanks.

**Comment by UdjinM6** [comment]
Should be fixed by #1739.
